extensions/ql-vscode/src/helpers.ts

```diff
@@ -266,6 +269,29 @@ export function isFolderAlreadyInWorkspace(folderName: string) {
   );
 }
 
+/** Check if the current workspace is the CodeTour and open the workspace folder.
+ * Without this, we can't run the code tour correctly.
+ **/
+export async function prepareCodeTour(): Promise<void> {
+  if (workspace.workspaceFolders?.length) {
+    const currentFolder = workspace.workspaceFolders[0].uri.fsPath;
+
+    const tutorialWorkspaceUri = Uri.parse(
+      join(workspace.workspaceFolders[0].uri.fsPath, "tutorial.code-workspace"),
+    );
+
+    const toursFolderPath = join(currentFolder, ".tours");
+
+    if (
+      existsSync(tutorialWorkspaceUri.fsPath) &&
+      existsSync(toursFolderPath) &&
+      !isCodespacesTemplate()
+    ) {
+      await commands.executeCommand("vscode.openFolder", tutorialWorkspaceUri);
```

Comment:
While this command is called `openFolder`, it actually accepts a workspace file URI and will open the workspace correctly. More details on this: https://github.com/Microsoft/vscode/issues/42610